cmd/e2d/app/run.go

```diff
@@ -52,6 +52,12 @@ type runOptions struct {
 	AWSSecretKey       string `env:"E2D_AWS_SECRET_KEY"`
 	AWSRoleSessionName string `env:"E2D_AWS_ROLE_SESSION_NAME"`
 
+	AzureAccountName    string        `env:"E2D_AZURE_ACCOUNT_NAME"`
+	AzureAccountKey     string        `env:"E2D_AZURE_ACCOUNT_KEY"`
```

Comment:
Azure recommends avoiding using Account Keys since they're not unique to a user, and instead using Managed Service Identities.

Unfortunately, like everything Azure, it's not super straightforward to use a MSI since there's multiple types of MSIs:
- System Managed Identities - these are assigned at instance start, and the wireserver will automatically issue this identity to you without any further configuration
- User Managed Identities - you can assign one or more user managed identities to a host, and when requesting tokens from the wireserver, you need to specify the UUID of the identity you wish to get credentials for if there are multiple user managed identities assigned to the host

It might be good to just add a `--azure-msi-id` flag to use a specific UUID if the system managed identity isn't sufficient.

There's no need to remove the account key flag, since some users (not us) may likely be using the legacy azure permission model (or are testing from their laptops!)

Reply:
Done.